### PR TITLE
Make the RealmSwift binary submittable to the iOS app store

### LINKED ISSUE #1907: Cannot submit an app built with RealmSwift to AppStore
I did installation steps in swift documentation.
added Framework Search Path as
$(PROJECT_DIR)/RealmSwift.framework/Frameworks

When I tried to submit to AppStore,

First, I got an error "Invalid Signature. Code object is not signed at all. The binary at path [MyApp.app/Frameworks/RealmSwift.framework/strip-frameworks.sh] contains an invalid signature. ...."

So I added following command in Build Phases > Run Script
rm ${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}/RealmSwift.framework/strip-frameworks.sh

Now I got this error.
"Invalid Bundle. The bundle at 'MyApp.app/Frameworks/RealmSwift.framework' contains disallowed nested bundles."
"Invalid Bundle. The bundle at 'MyApp.app/Frameworks/RealmSwift.framework' contains disallowed file 'Frameworks'."

Tested with Xcode 6.3.1, 6.3.2GM

Also tried:
Set Embedded Content Contains Swift Code to YES in Build Settings

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1907.

\c @jpsim @tgoyne @bdash

### CODE REVIEW

**Comment by tgoyne** [comment]
Even if nested frameworks work on OS X I think we may want to do the same thing there so that we don't need different install instructions for the two platforms.

Did you test building the full release package (`WORKSPACE=/tmp/whatever sh build.sh test-package-release`) and submitting to the app store with the built frameworks from that?

**Comment by segiddins** [comment]
No, but I did test that manually building the two frameworks (without using the now-deleted build phase script) was submittable to the app store)

**Comment by pietbrauer** [comment]
@segiddins @tgoyne I can confirm, I also build via `build.sh build` and it went through the app store submission.

**Comment by webmagnets** [comment]
What exactly do I need to type in to the terminal to try this fix?

**Comment by segiddins** [comment]
This has now been updated to also un-nest the OS X RealmSwift.framework, please re-review.

**Comment by jpsim** [comment]
Could you update the installation examples under `examples/installation` to reflect this change? We'll also need to update our installation instructions, so please draft a PR on realm.io once this is merged.

**Comment by jpsim** [comment]
Otherwise, :+1:

**Comment by segiddins** [comment]
The CI failure here is 
> 2015-05-26 13:15:04.703 xcodebuild[34272:468137] [MT] iPhoneSimulator: SimVerifier returned: Error Domain=NSPOSIXErrorDomain Code=53 "Simulator verification failed." UserInfo=0x7fcf8142ff60 {NSLocalizedFailureReason=A connection to the simulator verification service could not be established., NSLocalizedRecoverySuggestion=Ensure that Xcode.app is installed on a volume with ownership enabled., NSLocalizedDescription=Simulator verification failed.}

@tgoyne @bdash please give this a quick look over?

File: build.sh [hunk 1]
```diff
@@ -278,14 +278,16 @@ case "$COMMAND" in
         ;;
 
     "ios-dynamic")
-        build_combined "iOS Dynamic" Realm "-dynamic" "LD_DYLIB_INSTALL_NAME='@rpath/RealmSwift.framework/Frameworks/Realm.framework/Realm'"
+        build_combined "iOS Dynamic" Realm "-dynamic"
         exit 0
         ;;
 
     "ios-swift")
+        sh build.sh ios-dynamic
```
Comment (bdash) [inline build.sh]: Why is this now required?

**Comment by bdash** [comment]
:+1:

File: build.sh (hunk 1 above)
Comment (segiddins) [inline build.sh]: to ensure `build/ios-dynamic/Realm.framework` exists

Comment (bdash) [inline build.sh]: Why wasn't this step necessary prior to this change? Doesn't the shell script phase in RealmSwift's framework target already run `build.sh ios-dynamic` if the framework doesn't exist?

Comment (segiddins) [inline build.sh]: removed!

File: build.sh [hunk 2]
```diff
@@ -298,9 +299,8 @@ case "$COMMAND" in
         ;;
 
     "osx-swift")
+        sh build.sh osx
```
Comment (bdash) [inline build.sh]: Is this unnecessary for the same reason that the iso-dynamic one was?

Comment (segiddins) [inline build.sh]: right you are!

File: RealmSwift.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1033,9 +1012,8 @@
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				FRAMEWORK_SEARCH_PATHS = (
-					"$(SDKROOT)/Developer/Library/Frameworks",
 					"$(inherited)",
-					"$(CONFIGURATION_BUILD_DIR)/RealmSwift.framework/Frameworks",
+					"build/ios-dynamic",
```
Comment (tgoyne) [inline RealmSwift.xcodeproj/project.pbxproj]: Why "$(PROJECT_DIR)/build/ios-dynamic" above and "build/ios-dynamic" here?

Comment (segiddins) [inline RealmSwift.xcodeproj/project.pbxproj]: ¯\_(ツ)_/¯ I think Xcode expanded it in one place and not the other... I'll make it consistent

File: build.sh [hunk 4]
```diff
@@ -299,8 +300,6 @@ case "$COMMAND" in
 
     "osx-swift")
         xcrealmswift "-scheme 'RealmSwift OSX' -configuration $CONFIGURATION build"
-        rm -rf build/osx
```
Comment (tgoyne) [inline build.sh]: This does still need to be done to ensure that any files removed from the framework actually go away.

File: build.sh [hunk 5]
```diff
@@ -299,8 +300,6 @@ case "$COMMAND" in
...
-        mkdir build/osx
         cp -R build/DerivedData/RealmSwift/Build/Products/$CONFIGURATION/RealmSwift.framework build/osx
```
Comment (tgoyne) [inline build.sh]: Realm.framework doesn't have to be copied too?

File: build.sh (hunk 4 above)
Comment (segiddins) [inline build.sh]: it's done in `osx` which is invoked as target dependency of building `RealmSwift`

File: build.sh (hunk 5 above)
Comment (segiddins) [inline build.sh]: it's copied in `osx`

File: RealmSwift.xcodeproj/project.pbxproj (hunk 3 above)
Comment (segiddins) [inline RealmSwift.xcodeproj/project.pbxproj]: updated

**Comment by tgoyne** [comment]
:+1: 

CHANGELOG

**Pull request merged**